jwk.go

```diff
@@ -129,13 +130,13 @@ func (k JSONWebKey) MarshalJSON() ([]byte, error) {
 		if x5tSHA1Len != sha1.Size {
 			return nil, fmt.Errorf("square/go-jose: invalid SHA-1 thumbprint (must be %d bytes, not %d)", sha1.Size, x5tSHA1Len)
 		}
-		raw.X5tSHA1 = newFixedSizeBuffer(k.CertificateThumbprintSHA1, sha1.Size)
+		raw.X5tSHA1 = base64.RawURLEncoding.EncodeToString(k.CertificateThumbprintSHA1)
```

Comment:
This could be `newFixedSizeBuffer(k.CertificateThumbprintSHA1, sha1.Size).base64()` which I believe has additional error-checking and verifies the size.

Reply:
That causes some problems with !bytes.Equal on line 296 that's why I kept this one instead